pyfar/dsp/dsp.py

```diff
@@ -1487,4 +1495,194 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
             f"Domain is '{domain}', but has to be 'time', 'freq',"
             " or 'freq_raw'.")
     data[data == 0] = np.finfo(float).eps
-    return log_prefix * np.log10(np.abs(data) / log_reference)
+    if prefix_return is True:
+        return log_prefix * np.log10(np.abs(data) / log_reference), log_prefix
+    else:
+        return log_prefix * np.log10(np.abs(data) / log_reference)
+
+
+def normalize(signal, domain='time', dB=False, power=False, operation='max',
+              channel_handling='max', target=None, freq_range=None,
+              return_reference=False):
+    """
+    Apply a normalization in time or frequency domain.
+
+    In the default case, the normalization ensures that the maximum absolute
+    amplitude of the signal after normalization is 1. This is achieved by
+    the multiplication
+
+    ``signal_normalized = signal * target / reference``,
+
+    where `target` equals
+    1 and `reference` is the maximum absolute amplitude before the
+    normalization.
+
+    It is for example also possible normalize to the mean absolute amplitude or
+    to normalize the magnitude spectrum within a certain frequency range. This
+    is done as described above, but changes the way in which `reference` is
+    calculated.
+
+    Parameters
+    ----------
+    signal: Signal
+        Input signal of the signal class.
+    domain: string
+        The domain in which the signal is normalized
+
+        ``'time'``
+            Normalize the absolute time data.
+        ``'freq'``
+           Normalize the absolute frequency data. In this case the magnitude
+           spectrum with FFT normalization is used (``signal.freq``, cf.
+           :py:mod:`FFT concepts <pyfar._concepts.fft>`).
+
+        The default is ``'time'``.
+    dB: bool, optional
+        If `dB` is ``True``, the logarithmic magnitude spectra are used for
+        normalization. They are calculated by :py:func:`~pyfar.dsp.decibel`.
+        The default is ``False``.
+    power: bool, optional
+        If `power` is ``True``, the power spectra :math:`|X|^2` will be used
+        for normalization. Notice that `power` does have no effect if `dB` is
+        ``True``.
```

Comment:
According to the code, the data is also squared if `domain='time'`?
I'm not sure if this parameter is needed, aren't meaningful normalizations covered with ``operation='rms'`` or could atleast be covered by an additional operation ?
In contrast, this parameter leads to not meaningful results in case of some FFT normalization such as `fft_norm='power'` as well as in combination with ``operation='rms'``. What about the redundancy with the ``signal.fft_norm``?
So, let's list meaningful use cases here.